src/ofdm.c

```diff
@@ -746,29 +746,33 @@ static int est_timing(struct OFDM *ofdm, complex float *rx, int length,
         corr_st = 0.0f;
         corr_en = 0.0f;
 
-#if defined(__EMBEDDED__) && defined(__REAL__)
-        float re,im;
-
+#ifdef __EMBEDDED__
+#ifdef __REAL__
+        // Note: this code untested
+	float re,im;
+        
         codec2_dot_product_f32(&rx_real[i], wvec_pilot_real, ofdm->samplespersymbol, &re);
-        codec2_dot_product_f32(&rx_real[i], wvec_pilot_imag, ofdm->samplespersymbol, &im);
-        corr_st = re + im * I;
+	codec2_dot_product_f32(&rx_real[i], wvec_pilot_imag, ofdm->samplespersymbol, &im);
+	corr_st = re + im * I;
 
-        codec2_dot_product_f32(&rx_real[i+ ofdm->samplesperframe], wvec_pilot_real, ofdm->samplespersymbol, &re);
-        codec2_dot_product_f32(&rx_real[i+ ofdm->samplesperframe], wvec_pilot_imag, ofdm->samplespersymbol, &im);
-        corr_en = re + im * I;
-#elif defined(__EMBEDDED__)
```

Comment:
Wait, does the ARM version of gcc not like #elif for some reason or is this a style thing?

Reply:
Not sure.  When it wasn't working I started with master/ofdm.c and made small changes, so ended up with the original #if ... #endif structure.

Reply:
In theory both should be equivalent but if it's working now, it's not a big deal to keep as is.